### ISSUE TITLE
Exception being thrown all the time when using the light bottleneck without a connection

### ISSUE THREAD

**Issue by richtera** [SGrondin/bottleneck#127]
Although the exceptions themselves don't cause problems, they are very annoying when trying to debug and catch other exceptions. The this._store.heartbeat is not checked against null.
I think it would be better to do
```javascript
return typeof ((base = this._store.heartbeat) || {}).ref === "function" ? base.ref() : void 0;
```
or similar.
Or as in your code:
```coffeescript
    @_queues.on "leftzero", => @_store.heartbeat?.ref?()
    @_queues.on "zero", => @_store.heartbeat?.unref?()
```

![image](https://user-images.githubusercontent.com/708186/59511292-e4ae5100-8e83-11e9-83b2-2355b3df4315.png)

**Comment by richtera** [comment]
Sorry, I just moved this from the octokit/plugin-throttling.js#97 and closed the wrong one.

**Comment by SGrondin** [comment]
Wow this is very surprising to me. I'm shocked that it's possible and I'm shocked no one has reported it yet.

Could you please paste your octokit/plugin-throttle settings? Are you using Bottleneck directly at all or only through the plugin?

Thanks, I'll fix this ASAP.

**Comment by richtera** [comment]
It's a bug in bottleneck I think.
Basically the coffescript doesn't do null checking on "hearbeat" which is null for a light bottleneck without a connection. this._store.heartbear will be null, but it's trying to execute null.ref or null.unref. Ultimately the exception is caught in your even handler, but it's throwing unnecessary exceptions.

**Comment by richtera** [comment]
I tried both the default call from the plugin and passing bottleneck by importing it from 'bottleneck/light'. My settings just contain
```javascript
{
  onRateLimit: (retryAfter, options) => {
    winston.warn(`Request quote exhausted for request ${options.method} ${options.url}`);
    if (options.request.retryCount === 0) {
      winston.info(`Retrying after ${retryAfter} seconds`);
      return true;
    }
  },
  onAbuseLimit: (retryAfter, options) => {
    winston.warn(`Abuse detected for request $(opptions.method} ${options.url})`);
  },
}
```

**Comment by SGrondin** [comment]
Thanks, I'll release a fix this weekend

**Comment by SGrondin** [comment]
I'm sorry this took so long. I've been busy with another project and forgot about this issue. Thank you for your patience, it's fixed in [v2.19.2](https://github.com/SGrondin/bottleneck/releases/tag/v2.19.2)
